tuf/repository_tool.py

```diff
@@ -2669,135 +2645,150 @@ def remove_target_from_bin(self, target_filepath):
-        break
+  <Arguments>
+    target_filepath:
+      A path to a targetfile, relative to the targets directory
 
-      else:
-        logger.debug('"path_hash_prefix" not found.')
+  <Returns>
+    The hexdigest hash of the filepath.
+  """
 
-    # 'self._delegated_roles' is keyed by relative rolenames, so update
-    # 'hashed_bin_name'.
-    if hashed_bin_name is not None:
+  # TODO: ensure target_filepath is relative to targets_directory?
+  digest_object = securesystemslib.hash.digest(algorithm=HASH_FUNCTION)
+  digest_object.update(target_filepath.encode('utf-8'))
+  return digest_object.hexdigest()
 
-      # 'method_name' should be one of the supported methods of the Targets()
-      # class.
-      getattr(self._delegated_roles[hashed_bin_name], method_name)(target_filepath)
 
-    else:
-      raise securesystemslib.exceptions.Error(target_filepath + ' not found'
-          ' in any of the bins.')
 
 
+def _create_bin_name(low, high, prefix_len):
+  """
+  <Purpose>
+    Create a string name of a delegated hash bin, where name will be a range of
+    zero-padded (up to prefix_len) strings i.e. for low=00, high=07,
+    prefix_len=3 the returned name would be '000-007'.
 
-  @property
-  def delegations(self):
-    """
-    <Purpose>
-      A getter method that returns the delegations made by this Targets role.
+  """
+  if low == high:
+    return "{low:0{len}x}".format(low=low, len=prefix_len)
 
-      >>>
-      >>>
-      >>>
+  return "{low:0{len}x}-{high:0{len}x}".format(low=low, high=high,
+      len=prefix_len)
 
-    <Arguments>
-      None.
 
-    <Exceptions>
-      tuf.exceptions.UnknownRoleError, if this Targets' rolename
-      does not exist in 'tuf.roledb'.
 
-    <Side Effects>
-      None.
 
-    <Returns>
-      A list containing the Targets objects of this Targets' delegations.
-    """
 
-    return list(self._delegated_roles.values())
+def _get_bin_numbers(number_of_bins):
+  """
+  Given the desired number of bins (number_of_bins) calculate the prefix length
+  (prefix_length), total number of prefixes (prefix_count) and the number of
+  prefixes to be stored in each bin (bin_size).
+  Example: number_of_bins = 32
+    prefix_length = 2
+    prefix_count = 256
+    bin_size = 8
+  That is, each of the 32 hashed bins are responsible for 8 hash prefixes, i.e.
+  00-07, 08-0f, ..., f8-ff.
+  """
+  # Convert 'number_of_bins' to hexadecimal and determine the number of
+  # hexadecimal digits needed by each hash prefix
+  prefix_length = len("{:x}".format(number_of_bins - 1))
+  # Calculate the total number of hash prefixes (e.g., 000 - FFF total values)
+  prefix_count = 16 ** prefix_length
+  # Determine how many prefixes to assign to each bin
+  bin_size = prefix_count // number_of_bins
+
+  # For simplicity, ensure that 'prefix_count' (16 ^ n) can be evenly
+  # distributed over 'number_of_bins' (must be 2 ^ n).  Each bin will contain
+  # (prefix_count / number_of_bins) hash prefixes.
```

Comment:
Ping https://github.com/theupdateframework/tuf/pull/1007#discussion_r399269492

Reply:
Thanks for catching this. My original comment was lost when factoring out into the helper. I've updated that patch to add a comment as suggested.